proposals/XXXX-associated-types-constraints.md

```diff
@@ -0,0 +1,75 @@
+# Permit where clauses to constrain associated types
+
+* Proposal: [SE-XXXX](https://github.com/apple/swift-evolution/blob/master/proposals/XXXX-associated-types-constraints.md)
+* Author(s): [David Hart](http://github.com/hartbit), [Jacob Bandes-Storch](jtbandes@gmail.com), [Douglas Gregor](dgregor@apple.com)
+* Status: TBD
+* Review manager: TBD
+
+## Introduction
+
+This proposal seeks to introduce a `where` clause to associated types declarations and improvements to protocol constraints to bring associated types the same expressive power as generic types.
+
+This proposal was discussed twice on the Swift Evolution list in the following threads:
+
+- [[Completing Generics] Arbitrary requirements in protocols](http://thread.gmane.org/gmane.comp.lang.swift.evolution/14243)
+- [More Powerful Constraints for Associated Types](http://thread.gmane.org/gmane.comp.lang.swift.evolution/15201)
+ 
+## Motivation
+
+Currently, associated type declarations can only express simple inheritance constraints and not the more sophisticated constraints available to generic types with the `where` clause. Some designs, including many in the Standard Library, require more powerful constraints for associated types to be truly elegant. For example, the `SequenceType` protocol can be declared as follows:
```

Comment:
1. Please hard-wrap lines in markdown.  Otherwise diffs start to become very hard to work with.
2. s/can/could—if it were supported—/?